static/app/utils/duration/formatDuration.spec.tsx

```diff
@@ -77,34 +63,96 @@ describe('formatDuration', () => {
       ).toBe('8:20');
     });
 
-    it('should truncate ms when formatting as hours & minutes', () => {
+    it.each([
+      {duration: [500_000, 'ms'], precision: 'ms', expected: '8:20.000'},
+      {duration: [500_000, 'ms'], precision: 'sec', expected: '8:20.000'},
+      {duration: [500_012, 'ms'], precision: 'ms', expected: '8:20.012'},
+      {duration: [500_012, 'ms'], precision: 'sec', expected: '8:20.000'},
+      {duration: [500, 'sec'], precision: 'ms', expected: '8:20.000'},
+      {duration: [500, 'sec'], precision: 'sec', expected: '8:20.000'},
+      {duration: [500, 'sec'], precision: 'ms', expected: '8:20.000'},
+      {duration: [500, 'sec'], precision: 'sec', expected: '8:20.000'},
+      {duration: [500.012, 'sec'], precision: 'ms', expected: '8:20.012'},
+      {duration: [500.012, 'sec'], precision: 'sec', expected: '8:20.000'},
+      {duration: [500.012, 'sec'], precision: 'ms', expected: '8:20.012'},
+      {duration: [500.012, 'sec'], precision: 'sec', expected: '8:20.000'},
+    ] as Array<{duration: Duration; expected: string; precision: Unit}>)(
```

Comment:
i think there's a generic here instead of casting
```suggestion
    it.each<{duration: Duration; expected: string; precision: Unit}>([
      {duration: [500_000, 'ms'], precision: 'ms', expected: '8:20.000'},
      {duration: [500_000, 'ms'], precision: 'sec', expected: '8:20.000'},
      {duration: [500_012, 'ms'], precision: 'ms', expected: '8:20.012'},
      {duration: [500_012, 'ms'], precision: 'sec', expected: '8:20.000'},
      {duration: [500, 'sec'], precision: 'ms', expected: '8:20.000'},
      {duration: [500, 'sec'], precision: 'sec', expected: '8:20.000'},
      {duration: [500, 'sec'], precision: 'ms', expected: '8:20.000'},
      {duration: [500, 'sec'], precision: 'sec', expected: '8:20.000'},
      {duration: [500.012, 'sec'], precision: 'ms', expected: '8:20.012'},
      {duration: [500.012, 'sec'], precision: 'sec', expected: '8:20.000'},
      {duration: [500.012, 'sec'], precision: 'ms', expected: '8:20.012'},
      {duration: [500.012, 'sec'], precision: 'sec', expected: '8:20.000'},
    ])(
```